dbt/adapters/trino/connections.py

```diff
@@ -360,12 +361,24 @@ class TrinoConnectionManager(SQLConnectionManager):
     def exception_handler(self, sql):
         try:
             yield
-        # TODO: introspect into `DatabaseError`s and expose `errorName`,
-        # `errorType`, etc instead of stack traces full of garbage!
-        except Exception as exc:
-            logger.debug("Error while running:\n{}".format(sql))
-            logger.debug(exc)
-            raise dbt.exceptions.RuntimeException(str(exc))
+        except trino.exceptions.ProgrammingError as e:
```

Comment:
Can this be tested in any way?

Reply:
This might indeed be valuable to add a test for each 'type' of exception and ensure the correct dbt exception is raised. I will add that.

Reply:
Tests added